benches/README.md

```diff
@@ -0,0 +1,54 @@
+# Miden VM Hash Functions 
+In Miden VM, we make use of different hash functions. Some of these are "traditional" hash functions, like `Blake3`, which are optimized for out of STARK performance, while others are algebraic hash functions, like `Rescue Prime`, and are more optimized for a better performance inside the STARK. In what follows, we benchmark several such hash functions and compare against other constructions that are used by other proving systems. More precisely, we benchmark:
+
+## Rescue Prime
+As specified [here](https://eprint.iacr.org/2020/1143) and implemented [in](https://github.com/novifinancial/winterfell/blob/46dce1adf0/crypto/src/hash/rescue/rp64_256/mod.rs).
+
+## Rescue Prime Optimized
+As specified [here](https://eprint.iacr.org/2022/1577) and implemented in this crate.
+
+## BLAKE3
+As specified [here](https://github.com/BLAKE3-team/BLAKE3-specs/blob/master/blake3.pdf) and implemented in this crate.
+
+## SHA256
+As specified [here](https://csrc.nist.gov/csrc/media/publications/fips/180/4/final/documents/fips180-4-draft-aug2014.pdf) and implemented [in](https://github.com/novifinancial/winterfell/blob/46dce1adf0/crypto/src/hash/sha/mod.rs).
+
+## Poseidon
+As specified [here](https://eprint.iacr.org/2019/458.pdf) and implemented (without vectorized instructions) [in](https://github.com/mir-protocol/plonky2/blob/main/plonky2/src/hash/poseidon_goldilocks.rs).
+
+# Comparison and Instructions
+
+## Comparison
+We benchmark the above hash functions using two scenarios. The first is a 2-to-1 $(a,b)\mapsto h(a,b)$ hashing where both $a$, $b$ and $h(a,b)$ are $4$ field elements i.e. 256-bit.
+The second scenario is that of sequential hashing where we take a sequence of length $100$ field elements and hash these to produce a single digest.
+
+### Scenario 1: 2-to-1 hashing `h(a,b)` 
+
+| Function | Rp64_256 | RpO64_256 | BLAKE3 | SHA256 | Poseidon |
+| -------- | -------- | -------- | -------- | -------- | -------- |
+| Apple M1 | Text     | Text     | Text     | Text     | Text     |
+| AMD      | Text     | Text     | Text     | Text     | Text     |
+| x86 (CPU)| Text     | Text     | Text     | Text     | Text     |
+| ...      | Text     | Text     | Text     | Text     | Text     |
+
+### Scenario 2: Sequential hashing of 100 elements `h([a_0,...,a_99])`
+
+| Function | Rp64_256 | RpO64_256 | BLAKE3 | SHA256 | Poseidon |
+| -------- | -------- | -------- | -------- | -------- | -------- |
+| Apple M1 | Text     | Text     | Text     | Text     | Text     |
+| AMD      | Text     | Text     | Text     | Text     | Text     |
+| x86 (CPU)| Text     | Text     | Text     | Text     | Text     |
+| ...      | Text     | Text     | Text     | Text     | Text     |
+
+## Instructions
+To run these benchmarks, for RPO and Blake3, use
+
+```
+cargo bench
+```
```

Comment:
I think these could be clarified a bit more. E.g., we need to say that this should be run against this repo (so, maybe something like install rust, clone this repo, and then run the following command).

Also, this will run all benchmarks. Right now we have only hash benchmarks, but in the future we'll probably have others. So, probably good to limit this to hashing even now.

Reply:
Agree, let me know if the instructions are still lacking.